lib/endpoints/class-wp-rest-users-controller.php

```diff
@@ -92,6 +91,10 @@ public function get_items( $request ) {
 		$prepared_args['orderby'] = $orderby_possibles[ $request['orderby'] ];
 		$prepared_args['search'] = $request['search'];
 
+		if ( ! current_user_can( 'list_users' ) ) {
+			$prepared_args['has_published_posts'] = true;
```

Comment:
Should we only include posts marked as `show_in_rest`? Internally, WP grabs types with `public => true` here.